tests/unit/insights/test_importer.py

```diff
@@ -629,6 +640,53 @@ def generate_insights(
         batch_insert_mock.assert_called_once()
         product_insight_delete_mock.assert_called_once()
 
+    def test_import_msc_insights(self, mocker):
+        class FakeImporter(InsightImporter):
+            @staticmethod
+            def is_prediction_valid(
+                product: Product,
+                emb_code: str,
+            ):
+                return True
+
+            @classmethod
+            def generate_insights(cls, product, predictions):
+                yield [
+                    ProductInsight(
+                        barcode="4316268664431",
+                        type=InsightType.packager_code.name,
+                        value_tag="",
+                        value="ASC-C-00654",
+                        data={"raw": "asc-c-00654", "type": "fishing", "notify": False},
+                    )
+                ], [
+                    ProductInsight(
+                        barcode="0218978911697",
+                        type=InsightType.packager_code.name,
+                        value_tag="",
+                        value="ASC-C-00518",
+                        data={"raw": "asc-c-00518", "type": "fishing", "notify": False},
+                    )
+                ]
+
+        imported1 = FakeImporter.import_insights(
+            [Prediction(type="packager_code", value="ASC-C-00654")],
+            DEFAULT_SERVER_DOMAIN,
+            automatic=True,
+            product_store=FakeProductStore(),
+        )
```

Comment:
you don't need FakeImporter but want to use PackagerCodeInsightImporter

Moreover if you call import_insights it's hard to have a unit test, as it is writing to the database,
but if you look at it, you will see that it gets insights from generate_insights and that's what is important to you (you want to see if the right insights are generated), that's the function most of the test are testing in this file :-)

Your current test is inside the TestInsightImporter which is testing the generic InsightImporter, that's why it creates FakeImporters (because InsightImporter is not complete per se). This is not what you are testing.

Look instead, in this file there is already a TestPackagerCodeInsightImporter and it has a test_generate_candidates method !

So you just need to make a second method like this :-)
(or parametrize the existing test)